test/std-test.cc

```diff
@@ -95,6 +96,28 @@ TEST(std_test, optional) {
 #endif
 }
 
+// this struct exists only to force a variant to be
+// valueless by exception
+// NOLINTNEXTLINE
+struct throws_on_move_t {
+  throws_on_move_t() = default;
+
+  // NOLINTNEXTLINE
+  [[noreturn]] throws_on_move_t(throws_on_move_t&&) {
+    throw std::runtime_error{"Thrown by throws_on_move_t"};
+  }
+
+  throws_on_move_t(const throws_on_move_t&) = default;
+};
+
+template <> struct fmt::formatter<throws_on_move_t> : formatter<string_view> {
+  template <typename FormatContext>
+  auto format(const throws_on_move_t&, FormatContext& ctx) const {
```

Comment:
This doesn't need to be a template, you can use `format_context` instead.

Reply:
Fixed